lib/cli/src/Cardano/CLI.hs

```diff
@@ -1732,8 +1750,23 @@ walletStyleOption defaultStyle accepted = option (eitherReader fromTextS)
     prettyStyle s =
         "  " ++ T.unpack (toText s) ++ " (" ++ fmtAllowedWords s ++ ")"
 
+instance FromText (XPrvOrXPub -> Either String Text) where
```

Comment:
This is, interesting. Why not simply have an instance from `Text` to `Encoding` ? The encoding as a data-type is perfectly fine to carry on especially because the `encodeKey` arguments are homogeneous here. So this looks like needlessly complex.

Reply:
👍 Think I didn't have `KeyEncoding` in the same sense when I added this.